### ISSUE TITLE
Function block property categorization

### ISSUE THREAD

**Issue by e-grigorov** [eclipse/vorto#915]
I would like to trigger a discussion about this topic. Currently, the function block properties are put in one of these categories: status, configuration, fault or events. That's fine, but there are some side effects. For example, [Eclipse Ditto Integration](https://www.eclipse.org/ditto/basic-feature.html#the-link-to-eclipse-vorto) is going to push this:
```
"properties": {
  "configuration": {
    "on": true
  },
}
```
i.e. if there is a feature with "on" property, it must be put in one of the Vorto categories. We have Vorto specific data model(because of the category).

The alternative approach is;
```
"properties": {
  "on": true
}
```
but Vorto model cannot be defined for such feature.

Do you think that the Vorto property categorization can be relaxed to something like:
```
functionblock NewFunctionBlock {
	properties{
    ...
	}
	operations{
    ...
	}
}
```
or Ditto integration should be more flexible?

**Comment by aedelmann** [comment]
I don’t see the necessity to adapt Vorto to follow the exact same declaration as ditto because  Vorto is platform agnostic. It would also make the DSL even more harder to read and not as simple as now,  I feel. Plus this change would break down compatibility and would require a lot of changes in code generators etc. 
I actually like the simplicity of the Vorto language with less redundant key words but still descriptive enough to know what’s the meaning.

**Comment by Ebolon** [comment]
For me this is an topic to address in an generator. From my point of view it is a "misused concepts" that a Vorto model directly has an one to one representation in JSON. Vorto gives the data a meaning (types, constraints, etc.) but with a generated adapter your are free to define your format as you like. The Vorto model is maybe a bit to verbose. A generator for example could flatten the nesting level or minify the JSON. If you have an existing data source, you can not change, it require an adapter to map the data.
I am not 100% sure if this should be best-practice but you could restrict that property names cannot be used twice in configuration, status, etc. and remove this nesting level.

**Comment by e-grigorov** [comment]
> I don’t see the necessity to adapt Vorto to follow the exact same declaration as ditto because Vorto is platform agnostic.

Definitely, Vorto should stay platform agnostic and there is no need to adapt it to another model. The question is more about the current Vorto property categorization. This categorization is reused by Ditto.

> It would also make the DSL even more harder to read and not as simple as now, I feel. Plus this change would break down compatibility and would require a lot of changes in code generators etc. I actually like the simplicity of the Vorto language with less redundant key words but still descriptive enough to know what’s the meaning.

I agree that such a change will require many changes. That's why I started just a discussion. About the complexity, I have the feeling that declaration like:
```
properties {
...
}
operations {
...
}
```
is simpler than:
```
status {
...
}
configuration {
...
}
fault {
...
}
etc.
```

**Comment by e-grigorov** [comment]
> For me this is an topic to address in an generator. From my point of view it is a "misused concepts" that a Vorto model directly has an one to one representation in JSON. Vorto gives the data a meaning (types, constraints, etc.) but with a generated adapter you are free to define your format as you like. The Vorto model is maybe a bit to verbose. A generator for example could flatten the nesting level or minify the JSON. If you have an existing data source, you can not change, it require an adapter to map the data.

Ok, thanks. It answers to my question.

> I am not 100% sure if this should be best-practice but you could restrict that property names cannot be used twice in configuration, status, etc. and remove this nesting level.

Yes, good point.

**Comment by aedelmann** [comment]
Hi @e-grigorov 
How the Vorto Model is reflected in the Ditto structure should be a topic there and being discussed together with the Ditto team. 
Would you need any further clarifications about the Vorto categories and their semantics or can I close this issue ?

**Comment by kolotu** [comment]
Hi, @e-grigorov 
Status Properties are designed to represent read-only values. You can use the readable and writable flags to override that default behavior. 
However the proper place for writable properties is in the Configuration section - that is designed to represent properties that are readable and writable.

**Comment by kolotu** [comment]
The idea here was to change the naming of "status" and "configuration" to something that describes the readability/writeability for a future version of Vortolang (i.e. v2)
